templates/data-providers/index.jsx

```diff
@@ -0,0 +1,58 @@
+import React from 'react'
+import { TextField, Button, Card } from '@oacore/design'
+
+import styles from './styles.module.css'
+import SuccessMessage from './success-message'
+
+const DataProviderPageTemplate = React.forwardRef(
+  (
+    {
+      url,
+      onSubmit,
+      onUrlChange,
+      helperMessage,
+      variant,
+      showSuccessMessage,
+      oaiPmhUrl,
+      isFormValid = false,
+    },
+    ref
+  ) => {
+    const handleSubmit = (event) => {
+      event.preventDefault()
+      if (onSubmit) onSubmit(event)
+    }
+
+    return (
+      <Card className={styles.layout}>
+        {showSuccessMessage && <SuccessMessage oaiPmhUrl={oaiPmhUrl} />}
+        {!showSuccessMessage && (
+          <>
+            <Card.Title>Register new data provider</Card.Title>
+            <form ref={ref} className={styles.form} onSubmit={handleSubmit}>
+              <TextField
+                id="data-provider-url"
+                type="url"
+                name="dataProviderUrl"
+                label="Data provider URL"
+                helper={helperMessage}
+                placeholder="For example: oro.open.ac.uk"
+                value={url}
+                onChange={onUrlChange}
+                variant={variant}
+                statusIcon
+                required
+              />
+              <div className={styles.submitSection}>
+                <Button variant="contained" disabled={!isFormValid}>
```

Comment:
I still can submit if the button is disabled when press 'enter' in the field. `handleSubmit` should prevent if the validation is not passed or not finished.

Reply:
Interesting, it doesn't work for me with enter, you sure?

Reply:
Yes but I checked it in Safari, sorry. It does not work in Chrome. I wonder what's the difference

Reply:
It's sad, that safari works that way. https://github.com/oacore/search/pull/20/commits/2a4e3ce5a349b679d401c79ee4125085f7d25ca2 should fix it. Thanks for pointing it out.